File: practice_js/cpp_solutions/03-sum-nums.c
#include <stdio.h>
#include <stdlib.h>

long sum_nums(long num);

int main(){
    printf("The sum_nums of 5 is %ld \n", sum_nums(5));
    printf("The sum_nums of 10 is %ld \n", sum_nums(10));
    return 0;
}

long sum_nums(long num){
    if (num == 1){
        return 1;
    }
    else{
        return num + sum_nums(num-1);
    }
}

File: practice_js/javascript_solutions/03-sum-nums.js
//chain two functions together, the first creates a sequence of numbers, and the second reduces the array to a sum
function sumNums(number){
   return Array.from({length: number}, (el, index) => index + 1).reduce((sum,num) => {return sum = sum + num},0);
}


console.log(sumNums(5) == 15);
console.log(sumNums(4) == 10);
console.log(sumNums(3) == 6);
console.log(sumNums(8) == 36);
